nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/CommentUtils.java

```diff
@@ -100,20 +107,25 @@ public static void checkDocumentEventContext(Event event, DocumentModel expected
      */
     public static void checkReceivedMail(List<MailMessage> mails, DocumentModel commentDocModel,
             DocumentModel commentedDocModel, Event event, String commentEventType) {
+        DocumentModel sourceDocument = ((DocumentEventContext) event.getContext()).getSourceDocument();
         int expectedMailsCount = COMMENT_REMOVED.equals(commentEventType) ? 0 : 1;
+        expectedMailsCount = COMMENT_ADDED.equals(commentEventType)
+                && !sourceDocument.getId().equals(commentedDocModel.getId()) //
+                        ? 2
+                        : expectedMailsCount;
```

Comment:
You should leave ternary expression and use if/else if statements.

Reply:
Actually counting the mails doesn't do it in the case of replies. (Testing on replies is necessary to ensure the right document will be linked in the notification mail)
Tests around old comments flatly saved in the "Comments" folder fail because `NotificationEventListener#gatherConcernedUsersForDocumen`t will fail at gathering the subscribers in this folder:
```
private void gatherConcernedUsersForDocument(CoreSession coreSession, DocumentModel doc, List<Notification> notifs,
            Map<Notification, List<String>> targetUsers) {
        if (doc.getPath().segmentCount() > 1) {
            log.debug("Searching document: " + doc.getName());
            getInterstedUsers(doc, notifs, targetUsers);
            if (doc.getParentRef() != null && coreSession.exists(doc.getParentRef())) {
                DocumentModel parent = getDocumentParent(coreSession, doc);
                gatherConcernedUsersForDocument(coreSession, parent, notifs, targetUsers);
            }
        }
    }
```

I have removed mail counting in my last push force.

Reply:
I propose [another approach](https://github.com/nuxeo/nuxeo/pull/3587#discussion_r362759582)  to keep mail count.